internal/primitives/runtime/runtime.go

```diff
@@ -0,0 +1,37 @@
+// Copyright 2024 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package runtime
+
+// Justification is an abstraction over justification for a block's validity under a consensus algorithm.
+//
+// Essentially a finality proof. The exact formulation will vary between consensus algorithms. In the case where there
+// are multiple valid proofs, inclusion within the block itself would allow swapping justifications to change the
+// block's hash (and thus fork the chain). Sending a `Justification` alongside a block instead bypasses this problem.
+//
+// Each justification is provided as an encoded blob, and is tagged with an ID to identify the consensus engine that
+// generated the proof (we might have multiple justifications from different engines for the same block).
+type Justification struct {
```

Comment:
I think there is/will be overlapping types and logic between this and the grandpa branch that will need to be resolved

Reply:
I actually revisited a lot of the grandpa primitives to alias these ones in my `VoterWork` wip branch.